smithy-typescript-codegen/src/test/resources/software/amazon/smithy/typescript/codegen/test-map-with-sensitive-data.smithy

```diff
@@ -0,0 +1,25 @@
+namespace smithy.example
+
+@protocols([{name: "aws.rest-json-1.1"}])
+service Example {
+    version: "1.0.0",
+    operations: [GetFoo]
+}
+
+operation GetFoo(GetFooInput)
+
+structure GetFooInput {
+    foo: UserMap
+}
+
+map UserMap {
```

Comment:
It doesn't look like there are tests for the shape itself being marked sensitive for each of these types.

Reply:
The shape being marked sensitive is tested in:
* `filtersSensitiveStructure`
* `filtersSensitiveList`
* `filtersSensitiveMap`

Corresponding models code:

<details>
<summary>test-sensitive-structure.smithy</summary>

```
structure GetFooInput {
    @sensitive
    foo: User
}

structure User {
    username: String,

    @sensitive
    password: String
}
```

</details>

<details>
<summary>test-sensitive-list.smithy</summary>

```
structure GetFooInput {
    @sensitive
    foo: UserList
}

list UserList {
    member: User
}

structure User {
    username: String,

    @sensitive
    password: String
}
```

</details>

<details>
<summary>test-sensitive-map.smithy</summary>

```
structure GetFooInput {
    @sensitive
    foo: UserMap
}

map UserMap {
    key: String,
    value: User
}

structure User {
    username: String,

    @sensitive
    password: String
}
```

</details>

Reply:
Those tests are for sensitive members that point to the shapes in question, not for members that point to sensitive shapes:

```
structure GetFooInput {
     foo: UserMap
 }

@sensitive
map UserMap {
    key: String,
    value: User
}
```

Reply:
Modified existing tests to test sensitive shapes (list, map, structure) in [commit](https://github.com/awslabs/smithy-typescript/pull/170/commits/ad357b778af6ed14f4eb3e327a2bac3632113fc2)

Added following tests (although they are similar to `filtersSensitiveSimpleShape` in code path) in [commit](https://github.com/awslabs/smithy-typescript/pull/170/commits/3556c6454a7ebdc5b676a582639f9088abe8690b):
* `filtersSensitiveMemberPointingToStructure`
* `filtersSensitiveMemberPointingToList`
* `filtersSensitiveMemberPointingToMap`